src/Reflex/Query/Base.hs

```diff
@@ -291,8 +295,8 @@ withQueryT f a = do
   return result
 
 -- | Maps a function over a 'QueryT' that can change the underlying monad
-mapQueryT :: (forall b. m b -> n b) -> QueryT t q m a -> QueryT t q n a
-mapQueryT f (QueryT a) = QueryT $ mapStateT (mapEventWriterT (mapReaderT f)) a
+mapQueryT :: Monad m => (forall b. m b -> n b) -> QueryT t q m a -> QueryT t q n a
```

Comment:
This *technically* makes the change breaking...

Reply:
@3noch Do you think I should put it back? I can also make `hoist = mapQueryT`.

Reply:
`hoist = mapQueryT' might be better. The next release will be breaking anyway, but if there's a way to minimize it we might as well. I'm also not actually worried about this particular thing breaking anyone, but I have to err on the side of pedantic.

Reply:
Done.